cli/filelister.cpp

```diff
@@ -43,11 +43,16 @@
 static BOOL myIsDirectory(const std::string& path)
 {
 #ifdef __BORLANDC__
-    return (GetFileAttributes(path.c_str()) & FILE_ATTRIBUTE_DIRECTORY);
+    if (GetFileAttributes(path.c_str()) & FILE_ATTRIBUTE_DIRECTORY))
+        return TRUE;
 #else
 // See http://msdn.microsoft.com/en-us/library/bb773621(VS.85).aspx
-    return PathIsDirectoryA(path.c_str());
+    if (PathIsDirectoryA(path.c_str()))
+        return TRUE;
 #endif
+    else  if (endsWith(path, "*.cpp", 5) || endsWith(path, "*.c", 3))
```

Comment:
It looks wrong that `myIsDirectory` returns TRUE if there is a file extension.

Reply:
Also, `*` should be handled by the command line processor.. so how coud path end with `*.cpp`.